### PR TITLE
fix: auth init

### PR DESCRIPTION
**Pull request by gotoin**
hello, i found and fixed a bug.

if the this.mounted() throw an exception，this.$storage.watchState not be called

### CODE REVIEW

**Comment by gotoin** [comment]
config:
```
auth: {
    strategies: {
      local: {
        endpoints: {
          ...
        },
        tokenRequired: false, // set false
      },
    },
  }

```
./lib/schemes/local.js
```
  async fetchUser (endpoint) {
   ...
    
    // maybe the request return not 200 status
    const user = await this.$auth.requestWith(
        this.name,
        endpoint,
        this.options.endpoints.user
      )
    this.$auth.setUser(user)
  }

```

File: lib/core/auth.js [hunk 1]
```diff
@@ -44,17 +44,21 @@ export default class Auth {
...
-        if (!routeOption(this.ctx.route, 'auth', false)) {
-          this.redirect(loggedIn ? 'home' : 'logout')
-        }
-      })
+    
+    try {
+      // Call mounted for active strategy on initial load
+      await this.mounted()
+    } catch (e) {
+
```
Comment (manniL) [inline lib/core/auth.js]: we should handle the error here somehow 🤔

File: lib/core/auth.js [hunk 2]
```diff
@@ -44,17 +44,21 @@ export default class Auth {
...
+    } finally {
+      // Watch for loggedIn changes only in client side
+      if (process.browser && this.options.watchLoggedIn) {
```
Comment (manniL) [inline lib/core/auth.js]: Please use `process.client` (https://github.com/nuxt/nuxt.js/issues/4106)

**Comment by pi0** [comment]
This PR has been published in [v4.6.0](https://github.com/nuxt-community/auth-module/releases/tag/v4.6.0)

**Pull request merged**